nexus/reconfigurator/planning/src/planner.rs

```diff
@@ -1124,6 +1134,193 @@ impl<'a> Planner<'a> {
+                        "new_generation" => %new,
+                    );
+                } else {
+                    // It would be very strange for the current value to be
+                    // greater than the new value. That would indicate something
+                    // like a row being removed from the target release
+                    // generation table -- one of the invariants of the target
+                    // release generation is that it only moves forward.
+                    //
+                    // In this case we warn but set the value.
+                    warn!(
+                        log,
+                        "target release minimum generation was set to current, \
+                         but we are trying to set it to an older generation -- \
+                         this is unexpected and may indicate a problem with the \
+                         target release generation table. Setting the value \
+                         to the older generation anyway.";
+                        "current_generation" => %current,
+                        "new_generation" => %new,
+                    );
+                }
+                self.blueprint
+                    .set_target_release_minimum_generation(current, new)
+                    .expect("current value passed in => can't fail");
+            }
+        }
+
+        // Now we need to determine whether to also perform other actions like
+        // updating or adding zones. We have to be careful here:
+        //
+        // * We may have moved existing zones with an Artifact source to using
+        //   the install dataset via the BpSetOverride action, but we don't
+        //   want to use the install dataset on sleds that weren't mupdated
+        //   (because the mupdate might be ancient).
+        //
+        // * To be conservative, we don't want to start new zones on sleds
+        //   where any overrides are in place (XXX is this true? feels like we
+        //   may be able to get away with adding new zones on mupdated sleds?).
+        //
+        // This condition is level-triggered:
+        //
+        // * If any sleds have a mupdate override set in the blueprint,
+        //   then we're still recovering from a mupdate. If that is the case,
+        //   we don't want to perform further zone update actions.
+        // * If the planning input's target release generation is less than
+        //   the minimum generation set in the blueprint, the operator hasn't
+        //   set a new generation in the blueprint -- we should wait until they
+        //   do that.
+        //
+        // What does "any sleds" mean in this context? We don't need to care
+        // about decommissioned or expunged sleds, so we consider in-service
+        // sleds.
+
+        let mut reasons = Vec::new();
+        let mut sleds_with_override = BTreeSet::new();
+        for sled_id in self.input.all_sled_ids(SledFilter::InService) {
```

Comment:
Thinking about this, I'm wondering what would happen here if a sled goes away in the middle of this process, disappearing from inventory. In that case the remove_mupdate_override field never gets cleared from the blueprint.

We would do:

* expunge the sled in the first blueprint
* when executed, the sled policy will be updated to Expunged in the planning input
* 

So I think we'll eventually converge -- it'll just take a couple cycles.